gravitee-apim-console-webui/src/management/application/details/subscribe/application-subscribe.controller.ts

```diff
@@ -164,19 +164,24 @@ class ApplicationSubscribeController {
   shouldPromptForKeyMode(plan: any): boolean {
     return (
       plan.security === PlanSecurityType.API_KEY &&
-      this.allowsSharedApiKeys &&
+      this.isSharedApiKeyEnabled &&
+      this.application.api_key_mode === ApiKeyMode.UNSPECIFIED &&
       this.apiKeySubscriptionsCount >= 1 &&
-      this.application.api_key_mode === ApiKeyMode.UNSPECIFIED
+      this.canShareKey(plan)
     );
   }
 
   get apiKeySubscriptionsCount(): number {
     return this.subscriptions.data.filter((subscription) => subscription.security === PlanSecurityType.API_KEY).length;
   }
 
-  get allowsSharedApiKeys(): boolean {
+  get isSharedApiKeyEnabled(): boolean {
     return this.Constants.env?.settings?.plan?.security?.sharedApiKey?.enabled;
   }
+
+  canShareKey(plan: any): boolean {
```

Comment:
Could we improve readability with some kind of renaming ?
Suggestion : `hasAlreadySubscribedApiKeyPlanOnApi(api)`